docs/bring_your_own_model.md

```diff
@@ -0,0 +1,100 @@
+# Bring Your Own PyTorch Lightning Model
+
+The InnerEye toolbox is capable of training any PyTorch Lighting model inside of AzureML, making
+use of all the usual InnerEye toolbox features:
+- Working with different model in the same codebase, and selecting one by name
+- Distributed training in AzureML
+- Logging via AzureML's native capabilities
+- Training on a local GPU machine or inside of AzureML without code changes
+- Supply commandline overrides for model configuration elements, to quickly queue many jobs
+
+## Setup
+
+In order to use these capabilities, you need to implement a class deriving from `LightningContainer`. This class
+encapsulates everything that is needed for training with PyTorch Lightning:
+- The `create_model` method needs to return a subclass of `LightningModule`, that has
+all the usual PyTorch Lightning methods required for training, like the `training_step` and `forward` methods. This
+object needs to adhere to additional constraints, see below.
+- The `get_data_module` method needs to return a `LightningDataModule` that has the data loaders for
+training and validation data.
+- The optional `get_inference_data_module` returns a `LightningDataModule` that is used to read the data for inference
+(that is, evaluating the trained model). By default, this returns the same data as `get_training_data_module`, but you
+can override this for special models like segmentation models that are trained on equal sized image patches, but 
+evaluated on full images of varying size.
+
+Your class needs to be defined in a Python file in the `InnerEye/ML/configs` folder, otherwise it won't be picked up
+correctly. If you'd like to have your model defined in a different folder, please specify the Python namespace via
+the `--model_configs_namespace` argument. For example, use `--model_configs_namespace=My.Own.configs` if your
+model configuration classes reside in folder `My/Own/configs`.
+ 
+There are further requirements for the object returned by `create_model`, as described below.
+
+### Outputting files during training
+
+The Lightning model returned by `create_model` can inherit the fields defined in `OutputParams`.
```

Comment:
Should we replace "can inherit" by "can override"? Because Container always inherits from OutputParams no?